monai/metrics/cumulative_average.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from monai.metrics.utils import do_metric_reduction
+from monai.utils import MetricReduction
+
+from .metric import Cumulative
+
+
+class CumulativeAverage(Cumulative):
+    """
+    Cumulatively record data value and aggregate for the average value.
+    It supports single class or multi-class data, for example,
+    value can be 0.44 (like loss) or [0.3, 0.4] (like metrics of 2 classes).
+    It also supports distributed data parallel, sync data when aggregating.
+    For example, recording loss value and compute the oveall average value in every 5 iterations:
+
+    .. code-block:: python
+
+        average = CumulativeAverage()
+        for i, d in enumerate(dataloader):
+            loss = ...
+            average.append(loss)
+            if i % 5 == 0:
+                print(f"cumulative average of loss: {average.aggregate()}")
+        average.reset()
+
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.sum = None
+        self.not_nans = None
+
+    def reset(self):
+        """
+        Reset all the running status, including buffers, sum, not nans count, etc.
+
+        """
+        super().reset()
+        self.sum = None
+        self.not_nans = None
+
+    def aggregate(self):  # type: ignore
+        """
+        Sync data from all the ranks and compute the average value with previous sum value.
+
+        """
+        data = self.get_buffer()
+        if not isinstance(data, torch.Tensor):
+            raise ValueError("the data to aggregate must be PyTorch Tensor.")
+
+        # do metric reduction
+        f, not_nans = do_metric_reduction(data, reduction=MetricReduction.SUM_BATCH)
```

Comment:
this is the only line that the class is related to "metric", do we really need this in this class?

Reply:
Hi @wyli ,

Here we leveraged the reduction utility to compute SUM on not_nans, do you mean to compute here directly?
```py
not_nans = (~(torch.isnan(data))).float().sum(dim=0)
f = data.sum(dim=0)  # the batch sum
```

Thanks.

Reply:
Updated in the latest commit and added test for numpy data.

Thanks.